src/plugins/kibana/public/discover/components/field_chooser/field_chooser.js

```diff
@@ -188,7 +190,7 @@ define(function (require) {
             };
           }
 
-          $location.path('/visualize/create').search({
+          return $scope.state ? '#/visualize/create?' + $.param(_.assign($location.search(), {
```

Comment:
This would be easier to read as an early return. Eg on the first line of the function:

```
if (!$scope.state) return '';
```